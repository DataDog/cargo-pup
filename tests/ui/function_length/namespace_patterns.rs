//@compile-flags: --crate-name namespace_patterns
//@compile-flags: --crate-type lib

// This will not trigger an error, as it is in the root.
pub fn my_function() {
    let a = 1;
    let b = 2;
    let c = 3;
    let d = 4;
    let e = 5;
    let f = 6;
    let g = 7;
}

mod my_module {
    // This function should trigger the lint because it's in my_module
    pub fn my_module_function() { //~ ERROR: Function exceeds maximum length of 5 lines with 9 lines
        let a = 1;
        let b = 2;
        let c = 3;
        let d = 4;
        let e = 5;
        let f = 6;
        let g = 7;
    }

    // This nested module also matches the namespace pattern
    pub struct MyStruct;

    impl MyStruct {
        // This should trigger the lint as it's in my_module
        pub fn too_long_method(&self) { //~ ERROR: Function exceeds maximum length of 5 lines with 9 lines
            let a = 1;
            let b = 2;
            let c = 3;
            let d = 4;
            let e = 5;
            let f = 6;
            let g = 7;
        }
    }
}

mod other_module {
    // This function won't trigger the lint because it's in other_module
    pub fn other_module_function() {
        let a = 1;
        let b = 2;
        let c = 3;
        let d = 4;
        let e = 5;
        let f = 6;
        let g = 7;
    }
}