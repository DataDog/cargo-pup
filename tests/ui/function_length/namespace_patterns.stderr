error: Function exceeds maximum length of 5 lines with 9 lines
  --> tests/ui/function_length/namespace_patterns.rs:17:5
   |
LL | /     pub fn my_module_function() {
LL | |         let a = 1;
LL | |         let b = 2;
LL | |         let c = 3;
...  |
LL | |         let g = 7;
LL | |     }
   | |_____^
   |
   = help: 
   = note: Applied by cargo-pup rule 'namespace_patterns_test'.
   = note: `#[deny(pup_function_length)]` on by default

error: Function exceeds maximum length of 5 lines with 9 lines
  --> tests/ui/function_length/namespace_patterns.rs:32:9
   |
LL | /         pub fn too_long_method(&self) {
LL | |             let a = 1;
LL | |             let b = 2;
LL | |             let c = 3;
...  |
LL | |             let g = 7;
LL | |         }
   | |_________^
   |
   = help: 
   = note: Applied by cargo-pup rule 'namespace_patterns_test'.

error: aborting due to 2 previous errors

